openghg/store/base/_datasource.py

```diff
@@ -946,11 +825,11 @@ def keys_in_daterange_str(self, daterange: str) -> List[str]:
         start_date = timestamp_tzaware(split_daterange[0])
         end_date = timestamp_tzaware(split_daterange[1])
 
-        data_keys = self._data_keys["latest"]["keys"]
+        data_keys = self._data_keys[self._latest_version]
 
         return self.key_date_compare(keys=data_keys, start_date=start_date, end_date=end_date)
 
-    def key_date_compare(self, keys: Dict[str, str], start_date: Timestamp, end_date: Timestamp) -> List:
+    def key_date_compare(self, keys: List, start_date: Timestamp, end_date: Timestamp) -> List:
```

Comment:
I think `key_date_compare` is only used in `keys_in_daterange` and `keys_in_daterange_str`, and neither of those are used anywhere.

Reply:
OK I've opened an issue for this: https://github.com/openghg/openghg/issues/926